### PR TITLE
Made correct funcionality for checking region name

### PR DESCRIPTION
**Pull request by Tali32**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: src/pages/Regions/AddRegion.tsx [hunk 1]
```diff
@@ -108,165 +120,180 @@ const AddNewRegionFormPage = () => {
     setCurrentPhoto(false);
   };
 
+  function seeRegionNameExistsModal() {
```
Comment (ostappppppp) [inline src/pages/Regions/AddRegion.tsx:123]: What does it is mean `seeRegionNameExistsModal`?
I guess it should be `showCheckingRegionNameModal`

File: src/pages/Regions/AddRegion.tsx [hunk 2]
```diff
@@ -108,165 +120,180 @@ const AddNewRegionFormPage = () => {
...
+    return Modal.error({
+      title: "Округа з такою назвою вже існує! Будь-ласка вкажіть іншу назву.",
```
Comment (ostappppppp) [inline src/pages/Regions/AddRegion.tsx:125]: Typo here
```suggestion
      title: "Округа з такою назвою вже існує! Будь ласка, вкажіть іншу назву.",
```

File: src/pages/Regions/RegionEdit.tsx [hunk 3]
```diff
@@ -99,32 +100,51 @@ const RegionEditFormPage = () => {
...
-      postIndex: values.postIndex,
-      logo: logo,
-      city: values.city,
-      isActive: chosenRegion.isActive
-    };
-    await RegionsApi.EditRegion(currentRegion, newRegion);
+    setLoading(true);
 
-    form.resetFields();
+    const checkResponce = await checkIfNameExists(values.regionName);
```
Comment (ostappppppp) [inline src/pages/Regions/RegionEdit.tsx:105]: Does it really check some response, I guess it is checking just `regionName` for existing
So rename it please to something more appropriate

File: src/pages/Regions/RegionEdit.tsx [hunk 4]
```diff
@@ -99,32 +100,51 @@ const RegionEditFormPage = () => {
...
 
-    notificationLogic("success", successfulEditAction("Дані округи"));
-    history.push(`/regions/${currentRegion}`);
+    if (checkResponce.data === false) {
```
Comment (ostappppppp) [inline src/pages/Regions/RegionEdit.tsx:107]: Try to do something like:
```suggestion
    if (await checkIfNameExists(values.regionName).data) {
```

File: src/pages/Regions/RegionEdit.tsx [hunk 5]
```diff
@@ -99,32 +100,51 @@ const RegionEditFormPage = () => {
...
+      const newRegion: any = {
```
Comment (ostappppppp) [inline src/pages/Regions/RegionEdit.tsx:108]: Do you have any model for region instead of using `any`?

File: src/pages/Regions/RegionEdit.tsx [hunk 6]
```diff
@@ -99,32 +100,51 @@ const RegionEditFormPage = () => {
...
+      notificationLogic("success", successfulEditAction("Дані округи"));
+      history.push(`/regions/${currentRegion}`);  
+    } else {
+      setLoading(false);
+
+      seeRegionNameExistsModal();
+    }
   };
 
+  function seeRegionNameExistsModal() {
```
Comment (ostappppppp) [inline src/pages/Regions/RegionEdit.tsx:135]: Same here as I mentioned before
But also it is not modal for region but for `Округа`

File: src/pages/Regions/RegionEdit.tsx [hunk 7]
```diff
@@ -99,32 +100,51 @@ const RegionEditFormPage = () => {
...
+    return Modal.error({
+      title: "Округа з такою назвою вже існує! Будь-ласка вкажіть іншу назву.",
```
Comment (ostappppppp) [inline src/pages/Regions/RegionEdit.tsx:137]: Same here

**Review by ostappppppp — CHANGES_REQUESTED** [review]

File: src/pages/Regions/AddRegion.tsx (hunk 1 above)
Comment (Tali32) [reply src/pages/Regions/AddRegion.tsx:123]: It means to see the modal which says that region name already exists, not the modal which will check if the region name exist. So I could change it to "showRegionNameExistsModal"

File: src/pages/Regions/RegionEdit.tsx (hunk 5 above)
Comment (Tali32) [reply src/pages/Regions/RegionEdit.tsx:108]: Yes we have "RegionProfile" model

File: src/pages/Regions/RegionEdit.tsx (hunk 6 above)
Comment (Tali32) [reply src/pages/Regions/RegionEdit.tsx:135]: At the progect code we caled "Округа" as "Region"

File: src/pages/Regions/AddRegion.tsx (hunk 1 above)
Comment (Tali32) [reply src/pages/Regions/AddRegion.tsx]: Done!

File: src/pages/Regions/RegionEdit.tsx (hunk 4 above)
Comment (Tali32) [reply src/pages/Regions/RegionEdit.tsx]: Done!

File: src/pages/Regions/RegionEdit.tsx (hunk 3 above)
Comment (Tali32) [reply src/pages/Regions/RegionEdit.tsx]: Done!

File: src/pages/Regions/RegionEdit.tsx (hunk 5 above)
Comment (Tali32) [reply src/pages/Regions/RegionEdit.tsx]: Done!

File: src/pages/Regions/RegionEdit.tsx (hunk 6 above)
Comment (Tali32) [reply src/pages/Regions/RegionEdit.tsx]: Done!

File: src/pages/Regions/RegionEdit.tsx (hunk 7 above)
Comment (Tali32) [reply src/pages/Regions/RegionEdit.tsx]: Done!

File: src/pages/Regions/RegionEdit.tsx [hunk 8]
```diff
@@ -99,32 +100,50 @@ const RegionEditFormPage = () => {
...
+    } else {
+      setLoading(false);
 
-    notificationLogic("success", successfulEditAction("Дані округи"));
-    history.push(`/regions/${currentRegion}`);
+      showRegionNameExistsModal();
+    }
   };
 
+  function showRegionNameExistsModal() {
```
Comment (ostappppppp) [inline src/pages/Regions/RegionEdit.tsx:134]: You have exact the same method in `AddRegion.tsx`
You could move it to some common class for them both

**Review by sharix4 — APPROVED** [review]

File: src/pages/Regions/RegionEdit.tsx (hunk 8 above)
Comment (Tali32) [reply src/pages/Regions/RegionEdit.tsx:134]: we use methods of this type only to call the necessary modal windows in each component separately, so I don't think that it is necessary to move them to a separate class

Comment (ostappppppp) [reply src/pages/Regions/RegionEdit.tsx:134]: as you wish
but it still code duplicate😃

**Review by nicoolle — APPROVED** [review]

**Review by andrii52 — APPROVED** [review]

File: src/pages/Regions/RegionEdit.tsx (hunk 8 above)
Comment (Tali32) [reply src/pages/Regions/RegionEdit.tsx]: Removed code dublicate

**Review by ostappppppp — APPROVED** [review]

**Pull request merged**